validation/src/main/scala/hmda/validation/dsl/PredicateRegEx.scala

```diff
@@ -54,10 +54,10 @@ object PredicateRegEx {
 
   private def regExFor(pattern: String): String = {
     val result = pattern.map {
-      case 'N' => "[0-9]"
+      case 'N' => "\\d"
       case '.' => "\\."
     }
-    "^" + result.reduce(_ + _) + "$"
+    result.mkString("^", "", "$")
```

Comment:
I prefer this to the previous version. (Of course I do, or I wouldn't have made the change! Just commenting so it doesn't get lost in the noise.)